### PR TITLE
add documentation for PQS

### PR DESCRIPTION
**Pull request by hrischuk-da**

### CODE REVIEW

File: docs/2.7.1/docs/query/pqs-user-guide.rst [hunk 1]
```diff
@@ -271,6 +321,351 @@ PQS is able to start and finish at prescribed ledger offsets, specified by the a
...
+it is not guaranteed that these transactions will become visible to
+readers in order. The offset-based model makes the database’s isolation
+level irrelevant - so the loosest model (``read uncommitted``) is not
+harmful.
+
+The first thing to consider when querying the datastore is the type of
+read consistency required. If there is no need for consistency (eg.
+reading a historical contract - regardless of lifetime) then payload
+tables can be queried directly, without any consideration of offset.
+Another example could be a liveness metric query that calculates the
```
Comment (carrielaben-da) [inline docs/2.7.1/docs/query/pqs-user-guide.rst:604]:
```suggestion
Another example is a liveness metric query that calculates the
```

File: docs/2.7.1/docs/query/pqs-user-guide.rst [hunk 2]
```diff
@@ -271,6 +321,351 @@ PQS is able to start and finish at prescribed ledger offsets, specified by the a
...
+number of transactions in the datastore over the past minute. Again,
+this could be entirely valid without consideration of the
+parallel-writing method.
+
+When consistency is required, the reader must be aware of the offset
+from which they are reading. This will ensure they do not also read
```
Comment (carrielaben-da) [inline docs/2.7.1/docs/query/pqs-user-guide.rst:610]:
```suggestion
from which they are reading. This ensures they do not also read
```

File: docs/2.7.1/docs/query/pqs-user-guide.rst [hunk 3]
```diff
@@ -271,6 +321,351 @@ PQS is able to start and finish at prescribed ledger offsets, specified by the a
...
+further offsets that are present - but their precedent events are not yet
```
Comment (carrielaben-da) [inline docs/2.7.1/docs/query/pqs-user-guide.rst:611]:
```suggestion
further offsets that are present, but their precedent events are not yet
```

**Review by carrielaben-da — APPROVED** [review]
All changes should also apply to the 2.8 docs.

**Review by filmackay — APPROVED** [review]

**Pull request merged**